### ISSUE TITLE
Make pipelines useful for training

### ISSUE THREAD

**Issue by ozancaglayan** [pytorch/audio#2898]
### 🚀 The feature

Currently, the `pipelines` seem to be focusing solely on feature extraction purposes as it sets `model.eval()` internally (at least in Wav2Vec2 code that I quickly checked). Is there any particular reason for this, or is there anything that avoids me to use the returned module as part of another module and do training?

Also, to couple the feature request, I think there should also be an argument to the `bundle.get_model()` methods that disables downloading the original checkpoint as once its fine-tuned, there is no longer a need for the vanilla checkpoint.

Proposed API:
```python
self.backbone = bundle.get_model(training=True)
# train the model and save the weights with torch

# inference time
# training=False will skip setting .eval() and also should find a workaround for
# auto downloading of checkpoints from torch hub ideally.
self.backbone = bundle.get_model(training=False)

# load state dict of the full model which will bring weights associated to .backbone
```

### Motivation, pitch

It is often desirable to fine-tune a particular pre-trained model on downstream tasks so it would be good if this is recognised by torchaudio pipelines as well.

**Comment by mthrok** [comment]
Hi 

The pipeline bundles are primary designed for inference, but the design does not exclude its application for training. 

For eval mode vs train mode, wouldn't it be sufficient to call `train()` on the instantiated module?
For skipping the downloading and loading the pre-training weights, it is doable yet it kind of goes against the design principle. Would it make sense to use one of factory functions?

**Comment by ozancaglayan** [comment]
Hi,

Yes, it's sufficient to call `train()` on the module and it works fine for training. For the downloading part, if there is a band-aid solution that I could use that'd be great. When the model becomes part of another model, and you train it, it doesn't make sense to re-download the vanilla weights at all as they'll be simply overridden when model weights are loaded with load_state_dict.

**Comment by ozancaglayan** [comment]
https://github.com/pytorch/audio/blob/0be8423deb9c3546894f9082f8f7bdea265afa7b/torchaudio/pipelines/_wav2vec2/impl.py#L123

Looking at here, the code provides no way to skip this step. I can monkey-patch `_get_state_dict()` to return an empty dictionary but then `load_state_dict` obviously fails complaining about missing keys in the state dict.
